sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -579,6 +579,22 @@ class CodegenContext {
     s"${fullName}_$id"
   }
 
+  /**
+   * Creates an `ExprValue` representing a local java variable of required data type.
+   */
+  def freshName(name: String, dt: DataType): VariableValue = JavaCode.variable(freshName(name), dt)
+
+  /**
+   * Creates an `ExprValue` representing a local java variable of required data type.
+   */
+  def freshName(name: String, javaClass: Class[_]): VariableValue =
+    JavaCode.variable(freshName(name), javaClass)
+
+  /**
+   * Creates an `ExprValue` representing a local boolean java variable.
+   */
+  def isNullFreshName(name: String): VariableValue = JavaCode.isNullVariable(freshName(name))
```

Comment:
hmmm, now looking at this, `isNullFreshName(name)` doesn't save much typing compared to `freshName(name, BooleanType)`...

Reply:
haha, indeed. At least there is only one parameter. Maybe we should just use `freshName(name, BooleanType)`.

Reply:
yea, but we should open another PR to do this.

Reply:
change the new one in this PR but open another PR to rename `JavaCode.isNullVariable`

Reply:
Rename `JavaCode.isNullVariable`?

Reply:
`isNullFreshName` is new, we don't need it and can just call `freshName(name, BooleanType)`

Reply:
Ok.